allennlp/commands/train.py

```diff
@@ -257,7 +257,13 @@ def train_model(
         check_for_gpu(device_ids)
 
         master_addr = distributed_params.pop("master_address", "127.0.0.1")
-        master_port = distributed_params.pop("master_port", 29500)
+        if master_addr in ("127.0.0.1", "0.0.0.0", "localhost"):
+            # If running locally, we can automatically find an open port if one is not specified.
+            master_port = distributed_params.pop("master_port", common_util.find_open_port())
```

Comment:
This will run `find_open_port()` before it tries to read the port from the config. I think we should not run port discovery at all if we don't have to.

Reply:
Good point